### ISSUE TITLE
[Feature] Add better CSD for Linux

### ISSUE THREAD

**Issue by prateekmedia** [bitsdojo/bitsdojo_window#97]
Currently the window doesn't have any shadows and since flutter uses gtk decoration so you can render own shadows and by setting _GTK_FRAME_EXTENTS.

**Comment by knopp** [comment]
Can you be more specific? Unless you disable window decoration, the window is default Gtk window (using CSD on gnome and SSD on KDE) which should have shadows.

**Comment by prateekmedia** [comment]
Let me give you an example,

Take chrome, it has tabs in the header for linux, windows, mac, it is doing so by having a custom CSD, also it follows the turning of system on linux for header bar.

But currently if you want to add buttons to header bar you can't do that, you can only add them to appbar which looks awesome on android as it doesn't need window decoration but looks awkward for native desktop apps.

Another example would be MS office which adds File, Home and other options in top headerbar which looks more native.

Also in flutter if you remove the window decorations using bitsdojo window plugin  then you don't have the good things like drop shadows for window, rounded corners or even a native feel like It will not follow your current GTK / KDE theming.

To be clear I am talking about recreating below layout in flutter using nativeshell:

![](https://i.stack.imgur.com/knQ7T.png)

**Comment by knopp** [comment]
Sorry, accidentally replied this instead of the [NativeShell issue](https://github.com/nativeshell/nativeshell/issues/60). Can you reply there?

**Comment by bitsdojo** [comment]
Shadows and rounded corners are coming soon. Check this for a demo:

https://twitter.com/callmebit/status/1408040439609970703


![linux-rounded-shadow](https://user-images.githubusercontent.com/69184916/128490061-ee27d666-a68b-40fb-87e6-24475b3c753b.png)

**Comment by prateekmedia** [comment]
@bitsdojo awesome!

**Comment by prateekmedia** [comment]
@bitsdojo I was wondering if you can implement something like `WindowFrame.noTitle` which [nativeshell](https://github.com/nativeshell/nativeshell) has btw.

It helps you to get what your package does without removing the shadows and borders of original window.

What it basically does is removes only the headerbar and doesn't overwrite any other settings(Like shadows or borders).

**Comment by bitsdojo** [comment]
> I was wondering if you can implement something like `WindowFrame.noTitle`

Sure, it is on my list.

**Comment by ikeofkc** [comment]
this just got you a donation. thanks for all the work.

**Comment by 12people** [comment]
In the meantime, is there an interim fix that one could apply to get window shadows back?

**Comment by blemelin** [comment]
> In the meantime, is there an interim fix that one could apply to get window shadows back?

Yes. In `my_application.cc`, add this after `gtk_widget_show(GTK_WIDGET(window));` (around line 54).

```c++
auto gdk_window = gtk_widget_get_window(GTK_WIDGET(window));
gdk_window_set_decorations(gdk_window, GdkWMDecoration::GDK_DECOR_BORDER);
```
Not perfect, but it helps.

**Comment by diasnrg** [comment]
> > In the meantime, is there an interim fix that one could apply to get window shadows back?
> 
> Yes. In `my_application.cc`, add this after `gtk_widget_show(GTK_WIDGET(window));` (around line 54).
> 
> ```c++
> auto gdk_window = gtk_widget_get_window(GTK_WIDGET(window));
> gdk_window_set_decorations(gdk_window, GdkWMDecoration::GDK_DECOR_BORDER);
> ```
> 
> Not perfect, but it helps.

Does not work for Linux on Wayland windowing system, only for X11. Probably because Flutter supports X11.

**Comment by TaylorHo** [comment]
Do you happen to have any news here?

**Comment by prateekmedia** [comment]
@TaylorHo this package is almost dead, so either switch to window_manager plugin or wait for eternity.

**Comment by bitsdojo** [comment]
> Do you happen to have any news here?

@TaylorHo Looking to release a new version in August.

**Comment by annd22** [comment]
september now :(
